### PR TITLE
[process-agent] Add check subcommand and deprecate --check

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

- Add the `check` subcommand in the Process Agent replacing `--check` (`-check`).
- Only warn once if the `-version` flag is used.
- Deprecate the `--check` (`-check`) parameter (show warning on usage).

### Motivation

Improve consistency with the Core Agent

### Describe how to test/QA your changes

Test that the new `check` subcommand works correctly across supported OSs.

Since this touches the deprecation logic in these flags, they should be validated as well:
- `-config` (`--config`)
- `-sysprobe-config` (`--sysprobe-config`)
- `-version` (`--version`)
- `-check` (`--check`)
- `-pid` (`--pid`)
- `-info` (`--info`)
- `-install-service` (`--install-service`)
- `-uninstall-service` (`--uninstall-service`)
- `-start-service` (`--start-service`)
- `-stop-service` (`--stop-service`)
- `-foreground` (`--foreground`)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/process-agent/app/check.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	tagger.SetDefaultTagger(t)
+	err = tagger.Init(ctx)
+	if err != nil {
+		log.Errorf("failed to start the tagger: %s", err)
+	}
+	defer tagger.Stop() //nolint:errcheck
+
+	sysInfo, err := checks.CollectSystemInfo(cfg)
+	if err != nil {
+		log.Errorf("failed tp collect system info: %s", err)
```
Comment (wiyu) [inline cmd/process-agent/app/check.go:95]: Nit: typo?
```suggestion
		log.Errorf("failed to collect system info: %s", err)
```

File: cmd/process-agent/main_common.go [hunk 2]
```diff
@@ -101,44 +95,100 @@ func getSettingsClient(_ *cobra.Command, _ []string) (settings.Client, error) {
...
+		// Windows flags
+		"-install-service":   replaceFlagPosix,
+		"-uninstall-service": replaceFlagPosix,
+		"-start-service":     replaceFlagPosix,
+		"-stop-service":      replaceFlagPosix,
+		"-foreground":        replaceFlagPosix,
 	}
 
+	var newArgs []string
 	for i, arg := range args {
```
Comment (wiyu) [inline cmd/process-agent/main_common.go:173]: nit: You can remove the index var `i` here and use `arg` in the loop as it looks like you are no longer mutating the args slice anymore.

**Review by wiyu — APPROVED** [review]
👍 - left a few minor comments

File: cmd/process-agent/app/check.go (hunk 1 above)
Comment (xornivore) [reply cmd/process-agent/app/check.go:95]: Yes, thank you!

**Review by urseberry — APPROVED** [review]

**Review by mbotarro — APPROVED** [review]

File: cmd/process-agent/app/check.go [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	"github.com/DataDog/datadog-agent/pkg/tagger/remote"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/version"
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+)
+
+// CheckCmd is a command that runs the process-agent version data
+var CheckCmd = &cobra.Command{
+	Use:          "check",
+	Short:        "Run a specific check and print the results. Choose from: process, connections, realtime, process_discovery",
```
[the remark this answers is not in the document]
Comment (xornivore) [reply cmd/process-agent/app/check.go:34]: Yes, this is outdated, let me update.

File: cmd/process-agent/app/check.go [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+
+func runCheckCmd(cmd *cobra.Command, args []string) error {
+	// We need to load in the system probe environment variables before we load the config, otherwise an
+	// "Unknown environment variable" warning will show up whenever valid system probe environment variables are defined.
+	ddconfig.InitSystemProbeConfig(ddconfig.Datadog)
+
+	configPath := cmd.Flag(flags.CfgPath).Value.String()
+	sysprobePath := cmd.Flag(flags.SysProbeConfig).Value.String()
+	// `GetContainers` will panic when running in docker if the config hasn't called `DetectFeatures`.
+	// `LoadConfigIfExists` does the job of loading the config and calling `DetectFeatures` so that we can detect containers.
```
[the remark this answers is not in the document]
Comment (xornivore) [reply cmd/process-agent/app/check.go:50]: Sounds good, I will remove.

**Pull request merged**